backend_plugin/src/main/evaluators/BlockCommentEvaluator.java

```diff
@@ -3,114 +3,118 @@
+	 * Evaluates whether a warning should be triggered about the block comment feature
 	 */
 	public boolean evaluateDocumentChanges(DocumentEvent event) {
-		if (doubleBackSlash(event)) {
-			IDocument doc = event.getDocument();
-			try {
-				// Verify that the double slash is at the beginning of the line
-				// grab the position of the start of the line where the document change was
-				int lineOffset = doc.getLineInformationOfOffset(currentOffset).getOffset();
-				String textBeforeSlashes = doc.get(lineOffset, prevOffset-lineOffset);
-
-				textBeforeSlashes.trim();
-				if (textBeforeSlashes.isEmpty()) {
-
-					// If we have two back slashes, then check if we've found two previously
-					if (firstDoubleBackSlashDetected) {
-						IRegion currentRegion = event.getDocument().getLineInformationOfOffset(currentOffset);
-
-						// Check if consecutive lines
-						int prevToCurrentDif = currentRegion.getOffset() - (prevRegion.getOffset() + prevRegion.getLength());
-						int currentToPrevDif = prevRegion.getOffset() - (currentRegion.getOffset() + currentRegion.getLength());
-
-						// update the prevRegion
-						prevRegion = currentRegion;
-
-						// Uses a buffer of 2 characters because sometimes IRegion doesn't count the newly inserted "//"
-						if ((prevToCurrentDif >= 0 && prevToCurrentDif <= 2) ||
-								(currentToPrevDif >= 0 && currentToPrevDif <= 2)) {
-							return true;
-						}
-
-						// If we haven't seen a double back slash yet
-					} else {
-						firstDoubleBackSlashDetected = true;
-						prevRegion = event.getDocument().getLineInformationOfOffset(currentOffset);
-					}
+		try {
+
+			// Get the document the user is typing in
+			IDocument document = event.getDocument();
+
+			// Get the line number of the change
+			int line = document.getLineOfOffset(event.getOffset());
+
+			// Check if the line is now commented out after the change, but was not before the change
+			if (lineIsCommentedOut(document, line) && !lineWasPreviouslyCommentedOut(document, event, line)) {
+
+				// The user commented out a line. Check and see if they previously commented out an adjacent line
+				// manually
+				if (adjacentLineWasLastCommentedOut(line)) {
+					this.lastCommentedLine = line;
+					return true;
 				}
-			// Try Catch needed in case offset passed in doc.get methods doesn't actually exist
-			// We shouldn't actually get here, but since it's all asynchronous, something weird may happen
-			} catch (BadLocationException e) {
-				firstDoubleBackSlashDetected = false;
+
+				// Since the line has been commented out, update the information about the last commented line
+				this.lastCommentedLine = line;
+				this.lastCommentedLineTimeStamp = System.currentTimeMillis();
+
 			}
+		} catch (BadLocationException e) {
+			// This can happen in certain boundary positions (beginning and end) of the document. In these cases,
+			// just do nothing and return false
```

Comment:
Minor detail, but it's sort of confusing to have a comment here that says we return false, and then not return false until outside of the try-catch statement. I would just move the return statement in the catch block.

Reply:
I've updated the PR to place the return statement inside the catch.